### PR TITLE
feat: insurance paid blog new template

### PR DESCRIPTION
**Pull request by hero-dokane**
**Overview:** Create a new blog template specifically for articles that will be used for paid campaigns only. They will not show up in the blog directory and will not be indexed by search engines. 

Jira Ticket: https://pethealthinc.atlassian.net/browse/PM-273

Test URLs:
- Before: n / a
- After:
  - https://feature-insurance-paid-blog-template-new--petplace--hlxsites.hlx.page/insurance-paid-page
  - https://feature-insurance-paid-blog-template-new--petplace--hlxsites.hlx.page/insurance-paid-page/?martech=off

### CODE REVIEW

**Comment by ankyc-phi** [comment]
suggestion to get this out of drafts folder and put it on the root folder of petplace, would make it easier to test and deploy as well :)

**Comment by hero-dokane** [comment]
> suggestion to get this out of drafts folder and put it on the root folder of petplace, would make it easier to test and deploy as well :)

Confirming the page has been moved to the root folder of petplace: 

https://feature-insurance-paid-blog-template-new--petplace--hlxsites.hlx.page/insurance-paid-page

**Review by ankyc-phi — APPROVED** [review]

File: blocks/header/header.css [hunk 1]
```diff
@@ -37,6 +37,71 @@ header .header {
   position: fixed;
 }
 
+/* 
+* Header - Minimal 
+* Styling for the minimal header, which consists of the logo and the get a quote cta
+*/
+
+/* stylelint-disable-next-line selector-class-pattern */
+header:has(.header-content--minimal) {
```
Comment (ramboz) [inline blocks/header/header.css:46]: The `:has` pseudo selector still has some browser issues. I'd only use it for progressive enhancement, but make sure the style without it also works.

File: scripts/third-party.js
Comment (ramboz) [inline scripts/third-party.js:1]: You probably need to merge the main branch into the sprint deploy branch to get the latest changes we did and reduce the noise in the PR

File: search-index.db
Comment (ramboz) [inline search-index.db:1]: Should not be part of the PR. Likely same as above and needs to get the sprint deploy branch updated against main

File: templates/insurance-landing-page/insurance-landing-page.js
Comment (ramboz) [inline templates/insurance-landing-page/insurance-landing-page.js:1]: Same as above. sprint deploy branch needs to be udpated

File: blocks/header/header.js [hunk 2]
```diff
@@ -39,351 +36,427 @@ export default async function decorate(block) {
...
-
-    const html = await resp.text();
+    // Dev Note: this is a conditional check to use the minimal header
+    // for the insurance paid page template
+    if (window.location.href.includes('insurance-paid-page')) {
+      // build minimal header for insurance paid page, which contains the logo and a cta
+      navMeta = getMetadata('header-minimal');
+      navPath = navMeta
+        ? new URL(navMeta).pathname
+        : `${window.hlx.contentBasePath}/fragments/header-minimal`;
```
[the remark this answers is not in the document]
Comment (ramboz) [reply blocks/header/header.js:46]: I feel we need to clean up the header a bit… we used to have `nav` as a metadata, and then `mega-nav` was added, and now we have another `header-minimal`. I'd propose to rather simplify this and use a single `nav` metadata instead.
You can then create a conditional that checks for the fragment name in the given path to select between the 3 options if needed, but at least you simplify the authoring and avoid more confusion in the content.

[the remark this answers is not in the document]
Comment (ramboz) [reply blocks/header/header.js:46]: We could also go one step further and actually have 2 different optimized header blocks implementations instead so we only really load what's needed. I fear the 3 different headers are going to bloat the code quite a bit and we'll load a lot of JS/CSS when only 1/3 is really used.

You could have a generic `header` block that is used as a loader, and then selectively load `header-minimal`, `header-mega` & `header-default` based on the specified `nav` metadata and fragment URL.

We'd have clean code separation that way and reduce the risk of regressions down the road between the different header variants

File: templates/insurance-paid-page/insurance-paid-page.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  // same attribute setting as earlier
+  main.setAttribute('itemscope', '');
+}
+
+export async function loadLazy(document) {
+  const main = document.querySelector('main');
+  const heroTitleSection = document.createElement('div');
+  heroTitleSection.classList.add('hero-title-container', 'section');
+
+  const articleTitle = main.querySelectorAll('h1')[0];
```
Comment (ramboz) [inline templates/insurance-paid-page/insurance-paid-page.js:54]: if you do query the 1st anyway, you could just go with `.querySelector`

File: templates/insurance-paid-page/insurance-paid-page.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  const authorDiv = main.querySelector('.article-author-container');
+  authorDiv.classList.remove('section');
+  const heroImgContainer = main.querySelectorAll('p')[0];
+  heroImgContainer.classList.add('hero-pic-div');
+
+  heroTitleSection.append(articleTitle);
+  heroTitleSection.append(authorDiv);
+  heroTitleSection.append(heroImgContainer);
+  main.prepend(heroTitleSection);
```
Comment (ramboz) [inline templates/insurance-paid-page/insurance-paid-page.js:63]: I'd move that up in the eager phase. As it is you have content flicker (CLS) as you change the markup in the lazy phase (i.e. after it is shown to the end-user). You can see that in the PSI report

File: templates/insurance-paid-page/insurance-paid-page.js (hunk 3 above)
Comment (hero-dokane) [reply templates/insurance-paid-page/insurance-paid-page.js:54]: Hi @ramboz,  Are the header refactor updates required to be completed before we can merge this feature? Avinash asked if we can move these requested updates to a new ticket as a future enhancement.

Comment (ramboz) [reply templates/insurance-paid-page/insurance-paid-page.js:54]: Sure you can. It's just a suggestion. The one issue that concerns me more is the perf impact at the moment. A 39 perf score, even on a bad run, is really low. So I'd make sure to look into that as a first step. This will likely already require a bit of refactoring anyways.

**Comment by ankyc-phi** [comment]
@ramboz @hero-dokane we will be working on the tech debt for this PR after it's merged in, commenting here to have visibility
